dipy/denoise/noise_estimate.py

```diff
@@ -292,6 +331,6 @@ def estimate_sigma(arr, disable_background_masking=False):
     for i in range(sigma.size):
         convolve(arr[..., i], k, output=conv_out)
         mean_block = np.sqrt(6/7) * (arr[..., i] - 1/6 * conv_out)
-        sigma[i] = np.sqrt(np.mean(mean_block[mask]**2))
+        sigma[i] = np.sqrt(np.mean(mean_block[mask]**2) / factor)
```

Comment:
Yes: operators with higher precedence should be without white-space. But I
don't think this applies in cases where they are parenthesis-delimited,
such as here.

All the BDFL gave us mere mortals is the following guidance:

Yes:

i = i + 1
submitted += 1
x = x*2 - 1
hypot2 = x*x + y*y
c = (a+b) * (a-b)

No:

i=i+1
submitted +=1
x = x * 2 - 1
hypot2 = x * x + y * y

c = (a + b) * (a - b)

I don't know what to say, except that an automatic tool (like pep8) would
definitely flag this one. I agree with you that it doesn't read as nicely,
but that's the good thing about using standards: no one's individual taste
matters.


On Tue, May 19, 2015 at 6:03 PM, Samuel St-Jean <notifications@github.com>
wrote:

> In dipy/denoise/noise_estimate.py
> <https://github.com/nipy/dipy/pull/651#discussion_r30663829>:
>
> > @@ -292,6 +331,6 @@ def estimate_sigma(arr, disable_background_masking=False):
> >      for i in range(sigma.size):
> >          convolve(arr[..., i], k, output=conv_out)
> >          mean_block = np.sqrt(6/7) * (arr[..., i] - 1/6 * conv_out)
> > -        sigma[i] = np.sqrt(np.mean(mean_block[mask]**2))
> > +        sigma[i] = np.sqrt(np.mean(mean_block[mask]**2) / factor)
>
> Wasn't there a note saying you can keep the priority of operation grouped
> together like in
>
> a*2 + b*2
>
> Plus I really find it counterintuitive to have the exponent not close to
> the actual number, like in
>
> a ** 2 / b ** 2
> Seems messy :/
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/nipy/dipy/pull/651/files#r30663829>.
>